examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -88,27 +94,61 @@ namespace {
 // This file is being used by platforms other than Linux, so we need this check to disable related features since we only
 // implemented them on linux.
 #if CHIP_DEVICE_LAYER_TARGET_LINUX
+constexpr EndpointId kNetworkCommissioningEndpointMain      = 0;
+constexpr EndpointId kNetworkCommissioningEndpointSecondary = 0xFFFE;
+
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
 NetworkCommissioning::LinuxThreadDriver sLinuxThreadDriver;
-Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(0 /* Endpoint Id */, &sLinuxThreadDriver);
+Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(kNetworkCommissioningEndpointMain,
+                                                                             &sLinuxThreadDriver);
 #endif
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
 NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
-Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(1 /* Endpoint Id */, &sLinuxWiFiDriver);
-#endif
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(kNetworkCommissioningEndpointSecondary,
+                                                                           &sLinuxWiFiDriver);
 #endif
+#endif // CHIP_DEVICE_LAYER_TARGET_LINUX
 } // namespace
 
 void ApplicationInit()
 {
 #if CHIP_DEVICE_LAYER_TARGET_LINUX && defined(ZCL_USING_LEVEL_CONTROL_CLUSTER_SERVER)
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD && CHIP_DEVICE_CONFIG_ENABLE_WPA
+    emberAfEndpointEnableDisable(kNetworkCommissioningEndpointSecondary, false);
+
+    if (LinuxDeviceOptions::GetInstance().mThread && LinuxDeviceOptions::GetInstance().mWiFi)
+    {
+        sThreadNetworkCommissioningInstance.Init();
+        sWiFiNetworkCommissioningInstance.Init();
+        // Only enable secondary endpoint for network commissioning cluster when both WiFi and Thread are enabled.
+        emberAfEndpointEnableDisable(kNetworkCommissioningEndpointSecondary, true);
+    }
+    else
+#endif
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    sThreadNetworkCommissioningInstance.Init();
+        if (LinuxDeviceOptions::GetInstance().mThread)
+    {
+        sThreadNetworkCommissioningInstance.Init();
+    }
+    else
 #endif
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
-    sWiFiNetworkCommissioningInstance.Init();
-#endif
+        if (LinuxDeviceOptions::GetInstance().mWiFi)
+    {
+        // If we only enable WiFi on this device, "move" WiFi instance to main NetworkCommissioning cluster endpoint.
```

Comment:
So if ENABLE_WPA and not ENABLE_THREAD, we will not disable the secondary endpoint?   Shouldn't we disable it?

This nested compile-and-runtime conditional logic is pretty confusing....  How about:
```
  bool threadEnabled =
#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
    LinuxDeviceOptions::GetInstance().mThread;
#else
   false;
#endif
```
and similar for wifi, and then we can do:
```
  if (threadEnabled && wifiEnabled) {
  } else if (threadEnabled) {
    // Needs ifdef here around the Init() call, but if we reach here that define is always defined
  } else if (wifiEnabled) {
    // Similar
  } else {
    // Perhaps disable the secondary and set the primary to "Ethernet"?
  }
```

Reply:
Yes,  this is clearer, thanks.
also, moved `emberAfEndpointEnableDisable(kNetworkCommissioningEndpointSecondary, false);` to the outside of these if-s.